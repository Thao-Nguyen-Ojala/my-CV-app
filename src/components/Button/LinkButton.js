import React from 'react'
import Button from './Button';
import { Link } from 'react-router-dom';


function LinkButton({children, type, onClick, buttonStyle, buttonSize, buttonColor, directTo}) {

  return (
    <div>
      <Link to={`/${directTo}`}>
        <Button
          buttonColor={buttonColor}
          buttonSize={buttonSize}
          buttonStyle={buttonStyle}
          onClick={onClick}
          type={type}
          children={children}
        />
      </Link>
    </div>
  )
}

export default LinkButton
