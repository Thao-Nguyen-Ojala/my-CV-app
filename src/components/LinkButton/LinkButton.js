import React from 'react';
import Button from '../Button/Button';
import { Link } from 'react-router-dom';


function LinkButton({children, type, onClick, buttonStyle, buttonSize, buttonColor, directTo}) {

  return (
    <div>
      <Link data-testid='linkButton' to={`/${directTo}`}>
        <Button
          buttonColor={buttonColor}
          buttonSize={buttonSize}
          buttonStyle={buttonStyle}
          onClick={onClick}
          type={type}
          children={children}
        />
      </Link>
    </div>
  )
}

export default LinkButton
