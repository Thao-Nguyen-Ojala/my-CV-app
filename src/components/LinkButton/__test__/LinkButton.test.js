import React from 'react';
import ReactDOM from 'react-dom';
import LinkButton from '../LinkButton';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';


it('render LinkButton without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Router>
    <LinkButton>Link Button Testing</LinkButton>
    </Router>, div)
})

it('matches LinkButton snapshot', () => {
  const testingLinkButton = renderer.create(<Router><LinkButton className='btns' buttonStyle='btn--outline' buttonSize='btn--large'  buttonColor='red' directTo='testing'>Testing</LinkButton></Router>).toJSON();
  expect(testingLinkButton).toMatchSnapshot()
})