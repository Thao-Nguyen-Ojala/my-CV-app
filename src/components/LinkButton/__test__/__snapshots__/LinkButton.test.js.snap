// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches LinkButton snapshot 1`] = `
<div>
  <a
    data-testid="linkButton"
    href="/testing"
    onClick={[Function]}
  >
    <button
      className="btn btn--outline btn--large red"
      data-testid="button"
    >
      Testing
    </button>
  </a>
</div>
`;
